Release notes for v2.7 (#4400)
grafana/mimir: docs/sources/mimir/release-notes/v2.7.md

---
title: "Grafana Mimir version 2.7 release notes"
menuTitle: "V2.7 release notes"
description: "Release notes for Grafana Mimir version 2.7"
weight: 300
---

# Grafana Mimir version 2.7 release notes

Grafana Labs is excited to announce version 2.7 of Grafana Mimir.

The highlights that follow include the top features, enhancements, and bugfixes in this release. For the complete list of changes, see the [changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

## Features and enhancements

> - **Support for `keep_firing_for` option to ruler configuration**

Reviewer: What is this good for? Can we explain this feature a bit?

- **Experimental query sharding improvements**:
  A new configuration parameter, `-query-frontend.query-sharding-target-series-per-shard`, allows query sharding to take into account cardinality of similar requests executed previously when computing the maximum number of shards to use.
- **Experimental support for native histogram ingestion**:
  Native histograms can now be ingested. The new per-tenant limit `-ingester.native-histograms-ingestion-enabled` controls whether native histograms are stored or ignored.

### Alertmanager improvements

- **New metrics** The following upstream metrics are now exposed:
  - `cortex_alertmanager_dispatcher_aggregation_groups`
  - `cortex_alertmanager_dispatcher_alert_processing_duration_seconds`

### Helm chart improvements

> The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [corresponding documentation](/docs/helm-charts/mimir-distributed/latest/) for more information.

Reviewer 2: ```suggestion
The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [Grafana Mimir Helm chart documentation](/docs/helm-charts/mimir-distributed/latest/).
```

Reviewer 2: @krajorama, does this need to be a relref?

## Important changes

In Grafana Mimir 2.7, the default vaules of the following configuration options have changed:

> - `-blocks-storage.bucket-store.batch-series-size` is now enabled by default with a value of `5000`.

Reviewer 3: WDYT about mentioning this in the `Features and enhancements`? It's a fairly big change in the implementation and we've seen reduction in OOMs and memory usage for large requests. We've previously mentioned this as the "store-gateway streaming series results"

Author: Nice catch, I'm going to put this at the top of the new features and enhancements.

- `-blocks-storage.tsdb.retention-period` has changed from `24h` to `13h`.
- `-ruler.evaluation-delay-duration` has changed from `0` to `1m`.

In Grafana Mimir 2.7, the following configuration option is now deprecated:

- `-blocks-storage.bucket-store.chunks-cache.subrange-size` since there's no benefit to changing the default of `16000`

> In Grafana Mimir 2.7, the following have been removed:

Reviewer 2: ```suggestion
In Grafana Mimir 2.7, the following options, metrics, and labels have been removed:
```

> - Experimental support for ephemeral storage introduced in Mimir 2.6.0 has been removed.

Author: Yes, nice catch!

  - Following options are no longer available:
    - `-blocks-storage.ephemeral-tsdb.*`
    - `-distributor.ephemeral-series-enabled`
    - `-distributor.ephemeral-series-matchers`
    - `-ingester.max-ephemeral-series-per-user`
    - `-ingester.instance-limits.max-ephemeral-series`
  - The following metrics have been removed:
    - `cortex_ingester_ephemeral_series`
    - `cortex_ingester_ephemeral_series_created_total`
    - `cortex_ingester_ephemeral_series_removed_total`
    - `cortex_ingester_ingested_ephemeral_samples_total`
    - `cortex_ingester_ingested_ephemeral_samples_failures_total`
    - `cortex_ingester_memory_ephemeral_users`
    - `cortex_ingester_queries_ephemeral_total`
    - `cortex_ingester_queried_ephemeral_samples`
    - `cortex_ingester_queried_ephemeral_series`

>   - Additionally querying using the `{__mimir_storage__="ephemeral"}` selector no longer works. All label values with `ephemeral-` prefix in `reason` label of `cortex_discarded_samples_total` metric are no longer available.

Reviewer 2: ```suggestion
  - Additionally, querying using the `{__mimir_storage__="ephemeral"}` selector no longer works. All label values with the `ephemeral-` prefix within the `reason` label of the `cortex_discarded_samples_total` metric are no longer available.
```

## Bug fixes

> - Ingester: remove series from ephemeral storage even if there are no persistent series. [PR 4052](https://github.com/grafana/mimir/pull/4052)

Reviewer: No need to mention this bugfix, since entire ephemeral storage has been removed.

---
title: "Grafana Mimir version 2.7 release notes"
menuTitle: "V2.7 release notes"
description: "Release notes for Grafana Mimir version 2.7"
weight: 300
---

# Grafana Mimir version 2.7 release notes

Grafana Labs is excited to announce version 2.7 of Grafana Mimir.

The highlights that follow include the top features, enhancements, and bugfixes in this release. For the complete list of changes, see the [changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

> ## Features and enhancements

Reviewer 4: We should mention the following among new experimental features:

```
* [ENHANCEMENT] Store-gateway: use more efficient chunks fetching and caching. This should reduce CPU, memory utilization, and receive bandwidth of a store-gateway. Enable with `-blocks-storage.bucket-store.chunks-cache.fine-grained-chunks-caching-enabled=true`. #4163 #4174 #4227 #4255
```

> - **Store-gateway streaming enabled by default** The new default value of `5000` for `-blocks-storage.bucket-store.batch-series-size` enables store-gateway streaming in the default configuration. This means that series are loaded from object storage in batches rather than buffering them all in memory before returning to the querier. Enabling streaming can improve memory utilization in the store-gateway.

Reviewer 4: [nit] I would clarify what "improve memory utilization" means:

```suggestion
- **Store-gateway streaming enabled by default** The new default value of `5000` for `-blocks-storage.bucket-store.batch-series-size` enables store-gateway streaming in the default configuration. This means that series are loaded from object storage in batches rather than buffering them all in memory before returning to the querier. Enabling streaming can reduce memory utilization peaks in the store-gateway.
```

- **Support for `keep_firing_for` option to ruler configuration** This new option determines the amount of time an alert should keep firing while the ruler expression doesn't return results.
- **Experimental query sharding improvements**:

>   A new configuration parameter, `-query-frontend.query-sharding-target-series-per-shard`, allows query sharding to take into account cardinality of similar requests executed previously when computing the maximum number of shards to use.

Reviewer 4: ```suggestion
  A new configuration parameter, `-query-frontend.query-sharding-target-series-per-shard`, allows query sharding to take into account cardinality of similar requests executed previously when computing the maximum number of shards to use. If you want to try it out, we recommend starting with a value of `2500`.
```

- **Experimental support for native histogram ingestion**:

>   Native histograms can now be ingested. The new per-tenant limit `-ingester.native-histograms-ingestion-enabled` controls whether native histograms are stored or ignored.

Reviewer 4: ```suggestion
  Native histograms can now be ingested. The new per-tenant limit `-ingester.native-histograms-ingestion-enabled` controls whether native histograms are stored or ignored. The support for querying native histograms is not complete yet and it's expected to be available in the next release.
```

### Alertmanager improvements

- **New metrics** The following upstream metrics are now exposed:
  - `cortex_alertmanager_dispatcher_aggregation_groups`
  - `cortex_alertmanager_dispatcher_alert_processing_duration_seconds`

### Helm chart improvements

The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [Grafana Mimir Helm chart documentation](/docs/helm-charts/mimir-distributed/latest/).

## Important changes

In Grafana Mimir 2.7, the default vaules of the following configuration options have changed:

- `-blocks-storage.bucket-store.batch-series-size` is now enabled by default with a value of `5000`.
- `-blocks-storage.tsdb.retention-period` has changed from `24h` to `13h`.
- `-ruler.evaluation-delay-duration` has changed from `0` to `1m`.

In Grafana Mimir 2.7, the following configuration option is now deprecated:

> - `-blocks-storage.bucket-store.chunks-cache.subrange-size` since there's no benefit to changing the default of `16000`

Reviewer 4: Can also mention https://github.com/grafana/mimir/pull/4409 as deprecation, please?

- Experimental support for ephemeral storage introduced in Mimir 2.6.0 has been removed.
  - Following options are no longer available:
    - `-blocks-storage.ephemeral-tsdb.*`
    - `-distributor.ephemeral-series-enabled`
    - `-distributor.ephemeral-series-matchers`
    - `-ingester.max-ephemeral-series-per-user`
    - `-ingester.instance-limits.max-ephemeral-series`
  - The following metrics have been removed:
    - `cortex_ingester_ephemeral_series`
    - `cortex_ingester_ephemeral_series_created_total`
    - `cortex_ingester_ephemeral_series_removed_total`
    - `cortex_ingester_ingested_ephemeral_samples_total`
    - `cortex_ingester_ingested_ephemeral_samples_failures_total`
    - `cortex_ingester_memory_ephemeral_users`
    - `cortex_ingester_queries_ephemeral_total`
    - `cortex_ingester_queried_ephemeral_samples`
    - `cortex_ingester_queried_ephemeral_series`
  - Additionally, querying using the `{__mimir_storage__="ephemeral"}` selector no longer works. All label values with the `ephemeral-` prefix within the `reason` label of the `cortex_discarded_samples_total` metric are no longer available.

> - The store-gateway default index header reader no longer uses mmap and the mmap-based index header reader has been removed. The following flags have been changed:

Reviewer 4: I think this should be mentioned among features. In 2.6 the mmap-based index-header reader was still the default, so in this release we're effectively introducing the mmap-less index-header reader for whoever use the default config.

  - Additionally, querying using the `{__mimir_storage__="ephemeral"}` selector no longer works. All label values with the `ephemeral-` prefix within the `reason` label of the `cortex_discarded_samples_total` metric are no longer available.
- The store-gateway default index header reader no longer uses mmap and the mmap-based index header reader has been removed. The following flags have been changed:
  - `-blocks-storage.bucket-store.index-header.map-populate-enabled` has been removed
  - `-blocks-storage.bucket-store.index-header.stream-reader-enabled` has been removed
  - `-blocks-storage.bucket-store.index-header.stream-reader-max-idle-file-handles` has been renamed to `-blocks-storage.bucket-store.index-header.max-idle-file-handles`, and the corresponding configuration file option has been renamed from `stream_reader_max_idle_file_handles` to `max_idle_file_handles`

## Bug fixes

- Store-gateway: return Canceled rather than Aborted or Internal error when the calling querier cancels a label names or values request, and return Internal if processing the request fails for another reason. [PR 4061](https://github.com/grafana/mimir/pull/4061)

> - Fix JSON and YAML marshalling of ephemeral_series_matchers field in /runtime_config. [PR 4091](https://github.com/grafana/mimir/pull/4091)

Reviewer: No need to mention this bugfix, as entire feature was removed.

- **Experimental support for native histogram ingestion**:
  Native histograms can now be ingested. The new per-tenant limit `-ingester.native-histograms-ingestion-enabled` controls whether native histograms are stored or ignored. The support for querying native histograms is not complete yet and it's expected to be available in the next release.

### Alertmanager improvements

- **New metrics** The following upstream metrics are now exposed:
  - `cortex_alertmanager_dispatcher_aggregation_groups`
  - `cortex_alertmanager_dispatcher_alert_processing_duration_seconds`

### Helm chart improvements

The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [Grafana Mimir Helm chart documentation](/docs/helm-charts/mimir-distributed/latest/).

## Important changes

In Grafana Mimir 2.7, the default vaules of the following configuration options have changed:

- `-blocks-storage.bucket-store.batch-series-size` is now enabled by default with a value of `5000`.

> - `-blocks-storage.tsdb.retention-period` has changed from `24h` to `13h`.

Reviewer: This change comes from PR #4382, which isn't part of this release. We can either cherry-pick it, or remove this line.

Author: Thanks, I removed it
